### PR TITLE
[heft] Add CLI snapshot tests.

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

File: apps/heft/src/cli/HeftToolsCommandLineParser.ts [hunk 1]
```diff
@@ -189,13 +191,5 @@ export class HeftToolsCommandLineParser extends CommandLineParser {
       this._terminal.writeLine();
       this._terminal.writeErrorLine(error.stack!);
     }
-
-    await this._metricsCollector.flushAndTeardownAsync();
-
-    if (!process.exitCode || process.exitCode > 0) {
-      process.exit(process.exitCode);
```
Comment (octogonz) [inline apps/heft/src/cli/HeftToolsCommandLineParser.ts:196]: How does this not reintroduce the issue where a failed build returns a zero exit code?

Comment (iclanton) [reply apps/heft/src/cli/HeftToolsCommandLineParser.ts]: Fair enough. I can revert that commit.

File: common/reviews/api/node-core-library.api.md [hunk 2]
```diff
@@ -587,10 +588,10 @@ export class Sort {
 export class StringBufferTerminalProvider implements ITerminalProvider {
     constructor(supportsColor?: boolean);
     readonly eolCharacter: string;
-    getErrorOutput(): string;
-    getOutput(): string;
-    getVerbose(): string;
-    getWarningOutput(): string;
+    getErrorOutput(normalizeColorCodes?: boolean): string;
```
Comment (octogonz) [inline common/reviews/api/node-core-library.api.md:591]: @iclanton Should this be an `options` parameter?  We might need to add more options later.  

And `getErrorOutput(true)` isn't very explicit about what `true` means.

File: common/reviews/api/node-core-library.api.md [hunk 3]
```diff
@@ -587,10 +588,11 @@ export class Sort {
 export class StringBufferTerminalProvider implements ITerminalProvider {
     constructor(supportsColor?: boolean);
     readonly eolCharacter: string;
-    getErrorOutput(): string;
-    getOutput(): string;
-    getVerbose(): string;
-    getWarningOutput(): string;
+    getErrorOutput(options?: IStringBufferOutputOptions): string;
+    // Warning: (ae-forgotten-export) The symbol "IStringBufferOutputOptions" needs to be exported by the entry point index.d.ts
```
Comment (octogonz) [inline common/reviews/api/node-core-library.api.md:592]: This needs to be included in the API

**Pull request merged**